checkov/common/graph/db_connectors/db_connector.py

```diff
@@ -1,12 +1,22 @@
-class DBConnector:
-    def save_graph(self, local_graph):
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, TypeVar, Generic
+
+if TYPE_CHECKING:
+    from checkov.common.graph.graph_builder.local_graph import LocalGraph
+
+T = TypeVar("T")
```

Comment:
Why do we need it?

Reply:
Because `DBConnector` is a generic class, so when someone subclasses it, he can just add the type, which will be returned by the `get_reader_endpoint` and `get_writer_endpoint`